File: java/primes/Simple.java
package primes;

public final class Simple {
    private static final int NUMBER = 1000000;

    public static int printPrimes(final int count) {
        int lastPrime = 2;

        for (var i = 3; i <= count; i += 2) {
            boolean isPrime = true;
            int limit = (int) Math.sqrt(i);

            for (var j = 3; j <= limit; j += 2) {
                if (i % j == 0) {
                    isPrime = false;
                    break;
                }
            }

            if (isPrime) {
                lastPrime = i;
            }
        }

        return lastPrime;
    }

    public static void main(String[] args) {
        long start = System.currentTimeMillis();

        long lastPrime = printPrimes(NUMBER);
        System.out.println("Last prime: " + lastPrime);

        long end = System.currentTimeMillis();
        long duration = end - start;

        System.out.println("Execution time: " + duration + "ms");
    }
}

File: zig/primes/Simple.zig
const std = @import("std");

const NUMBER = 1000000;

fn getLastPrime(count: i32) u32 {
    var lastPrime: u32 = 0;

    var i: u32 = 3;
    while (i <= count) : (i += 2) {
        var isPrime = true;
        const limit: u32 = @intCast(std.math.sqrt(i));

        var j: u32 = 3;
        while (j <= limit) : (j += 2) {
            if (@mod(i, j) == 0) {
                isPrime = false;
                break;
            }
        }

        if (isPrime) {
            lastPrime = i;
        }
    }

    return lastPrime;
}

pub fn main() anyerror!void {
    const stdout = std.io.getStdOut().writer();
    
    const start_time = std.time.milliTimestamp();

    const last_prime = getLastPrime(NUMBER);
    try stdout.print("Last prime: {}\n", .{last_prime});

    const end_time = std.time.milliTimestamp();
    const duration = end_time - start_time;

    try stdout.print("Execution time: {}ms\n", .{duration});
}
